runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPanel.java

```diff
@@ -410,36 +411,51 @@ public void focusLost(FocusEvent e)
 				String existing = configManager.getConfiguration(cd.getGroup().value(), cid.getItem().keyName());
 
 				Color existingColor;
-				JButton colorPickerBtn;
+				ColorJButton colorPickerBtn;
+
+				boolean alphaHidden = cid.getAlpha() == null;
 
 				if (existing == null)
 				{
 					existingColor = Color.BLACK;
-					colorPickerBtn = new JButton("Pick a color");
+					colorPickerBtn = new ColorJButton("Pick a color");
 				}
 				else
 				{
 					existingColor = ColorUtil.fromString(existing);
-					colorPickerBtn = new JButton(ColorUtil.toHexColor(existingColor).toUpperCase());
+					String colorHex = ColorUtil.toHexColor(existingColor).toUpperCase();
+					if (!alphaHidden)
+					{
+						colorHex = ColorUtil.toAlphaHexColor(existingColor).toUpperCase();
+					}
+
+					colorPickerBtn = new ColorJButton(colorHex);
```

Comment:
Why is this not just `colorPickerBtn = new ColorJButton(existing);`?

Reply:
`existing` seems to be an int representing the colour (like `-16711936` for colour `#ff00ff00`), but the button text that the constructor expects is a hex-encoded color like #ff00ff00. Do you mean to move the `ColorUtil.to[Alpha]HexColor(…)` calls and surrounding logic to the constructor and let it take an `int color` parameter instead?

Reply:
It just seems strange due to the string -> int -> string conversion. But I see it is to get the hex-encoded value of the integer. Maybe you can just change this to `Color existing = configManager.getConfiguration(cd.getGroup().value(), cid.getItem().keyName(), Color.class);` and then do the conversion from that to a hex string which would look better.

Reply:
That overloaded getConfiguration is nifty, I'll use that :+1: